docs/architecture/maui/unit-testing.md

```diff
@@ -1,9 +1,223 @@
+        : base(dialogService, navigationService, settingsService)
+    {
+        _appEnvironmentService = appEnvironmentService;
+        _settingsService = settingsService;
+    }
+}
+```
+
+The `OrderDetailViewModel` class has a dependency on the `IAppEnvironmentService` type, which the dependency injection container resolves when it instantiates an `OrderDetailViewModel` object. However, rather than create an `IAppEnvironmentService` object which utilizes real servers, devices and configurations to unit test the `OrderDetailViewModel` class, instead, replace the `IAppEnvironmentService` object with a mock for the purpose of the tests.
+
+This approach allows the `IAppEnvironmentService` object to be passed into the `OrderDetailViewModel` class at runtime, and in the interests of testability, it allows a mock class to be passed into the `OrderDetailViewModel` class at test time. The main advantage of this approach is that it enables unit tests to be executed without requiring unwieldy resources such as runtime platform features, web services, or databases.
+
+## Testing MVVM applications
+
+Testing models and view models from MVVM applications is identical to testing any other classes, and uses the same tools and techniques; this includes features such as unit testing and mocking. However, some patterns that are typical to model and view model classes can benefit from specific unit testing techniques.
+
+> [!TIP]
+> Test one thing with each unit test.
+
+Don't be tempted to make a unit test exercise more than one aspect of the unit's behavior. Doing so leads to tests that are difficult to read and update. It can also lead to confusion when interpreting a failure.
+
+The eShopOnContainers mobile app uses [xUnit](https://xunit.github.io/) to perform unit testing, which supports two different types of unit tests:
+
+|Testing Type | Description |
+|---------|---------|
+| Facts | Tests that are always true, which test invariant conditions |
+| Theories | Tests that are only true for a particular set of data. |
+
+The unit tests included with the eShopOnContainers mobile app are fact tests, so each unit test method is decorated with the `[Fact]` attribute.
+
+## Testing asynchronous functionality
+
+When implementing the MVVM pattern, view models usually invoke operations on services, often asynchronously. Tests for code that invokes these operations typically use mocks as replacements for the actual services. The following code example demonstrates testing asynchronous functionality by passing a mock service into a view model:
+
+```csharp
+[Fact]
+public async Task OrderPropertyIsNotNullAfterViewModelInitializationTest()
+{
+    var orderService = new OrderMockService();
+    var orderViewModel = new OrderDetailViewModel(orderService);
+
+    var order = await orderService.GetOrderAsync(1, GlobalSetting.Instance.AuthToken);
+    await orderViewModel.InitializeAsync(order);
+
+    Assert.NotNull(orderViewModel.Order);
```

Comment:
```suggestion
    // Arrange
    var orderService = new OrderMockService();
    var orderViewModel = new OrderDetailViewModel(orderService);

    // Act
    var order = await orderService.GetOrderAsync(1, GlobalSetting.Instance.AuthToken);
    await orderViewModel.InitializeAsync(order);

    // Assert
    Assert.NotNull(orderViewModel.Order);
```

Maybe have these comments here for the first one, with a note that establishes (and reaffirms this pattern from early mentions in this chapter), but states that they're not needed when the pattern is simple like this.

Reply:
I like the way that this reads far better.